patches/tModLoader/Terraria/ModLoader/Core/ModCompile.cs

```diff
@@ -170,13 +171,18 @@ internal static void BuildModCommandLine(string modFolder)
 			new ModCompile(new ConsoleBuildStatus()).Build(modFolder);
 		}
 		catch (BuildException e) {
-			Console.Error.WriteLine("Error: " + e.Message);
+			ErrorReporting.LogStandardDiagnosticError(e.Message, ErrorReporting.TMLErrorCode.TML002);
 			if (e.InnerException != null)
 				Console.Error.WriteLine(e.InnerException);
 			Environment.Exit(1);
 		}
 		catch (Exception e) {
-			Console.Error.WriteLine(e);
+			if (e is IOException iOException) {
+				ErrorReporting.LogStandardDiagnosticError("Please close tModLoader or disable the mod in-game to build mods directly.", ErrorReporting.TMLErrorCode.TML001);
+			}
```

Comment:
I imagine this exception is thrown during `TmodFile.Save`. It would be better to report it there. Perhaps we can add an optional error code to `BuildException`

Reply:
Well, the error message is specifically relating to building the mod directly not in-game, so I figure having it here in `BuildModCommandLine` makes sense. I can put it in `TmodFile.Save` though, it's just a bit ugly try-catching the `using (fileStream = File.Create(path))` line since I'd have to wrap the whole scope of `fileStream`.

Reply:
Perhaps wrap the `Save` call and theow a custom exception?